app.rb

```diff
@@ -0,0 +1,252 @@
+    age = gets.chomp.to_i
+
+    print 'Enter the student\'s name: '
+    name = gets.chomp
+
+    parent_permission = false
+    print 'Has parent permission? [Y/N]: '
+    input = gets.chomp.downcase
+    parent_permission = true if input == 'y'
+
+    student = Student.new(age, @class, name, parent_permission)
+    @person << student
+    puts 'Student created!'
+    sleep 1
+    menu
+  end
+
+  def create_a_teacher
+    print 'Enter the teacher\'s age: '
+    age = gets.chomp.to_i
+
+    print 'Enter the teacher\'s name: '
+    name = gets.chomp
+
+    print 'specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name)
+    @person << teacher
+    puts 'Teacher created!'
+    sleep 1
+    menu
+  end
+
+  def create_book
+    puts `clear`
+    puts 'Book Creation'
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @book << book
+
+    puts "#{book.title} added successfully!"
+    sleep 1
+    menu
+  end
+
+  def create_rental
+    if @book.empty?
+      puts 'No book has been created! Please create a book first.'
+      sleep 2
+      menu
+    end
+    puts 'Select a book from the following list by number'
+    @book.each_with_index do |book, index|
+      puts "#{index + 1}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+
+    if book_index > @book.length
+      puts 'Invalid input! Please enter a number between 1 and the number of books.'
+      sleep 1
+      create_rental
+    end
+
+    if @person.empty?
+      puts 'No person has been created! Please create a person first.'
+      sleep 2
+      menu
+    end
+
+    puts 'Select a person from the following list by index number (not id)'
+    @person.each_with_index do |person, index|
+      puts "#{index + 1}) [#{person.class}] Name: #{person.name}, id: #{person.id} Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+
+    if person_index > @person.length
+      puts 'Invalid input! Please enter a number between 1 and the number of people.'
+      sleep 1
+      create_rental
+    end
+
+    print 'Date: '
+    date = gets.chomp
+    date = date.empty? ? DateTime.now : date
+    rental = Rental.new(@person[person_index - 1], @book[book_index - 1], date)
+    @rental << rental
+
+    puts 'Rental created successfully!'
+    sleep 1
+    menu
+  end
+
+  def list_rentals_by_person_id
+    puts 'ID of the person: '
+    person_index = gets.chomp.to_i
+
+    if @rental.empty?
+      puts 'There are no rentals for this person'
+    else
+      @rental.each do |rental|
+        if rental.person.id == person_index
+          puts "#{rental.book.title} rented by #{rental.person.name} on #{rental.date}"
+        else
+          puts 'There are no rentals for this person'
+        end
+      end
+    end
+    puts 'Press any key to go back to the main menu....'
+    gets.chomp
+    menu
+  end
```

Comment:
- [ ] (Optional) When I list a rental for a given person using the `ID`, I realize it shows the book rented for the person and this string `There are no rentals for this person`. It would be nice to show the name of the book rented if the person has rented a book or the string if the person has not rented a book.

![image](https://user-images.githubusercontent.com/77106962/163673019-2f386cf3-60e6-4071-9819-35811ec44e8f.png)

Reply:
Thank you. I will make sure i implement that

Reply:
I just tried fixing the problem But I discovered the line of text only shows on my side if you enter a wrong ID. I used an if/else statement and it is working well here. Please can you test the app again and do you have any suggestions as to what I can do if you still get the same results?

Reply:
Hi @fobadara , I attached the screenshot to show you the output I got. It is optional, but it would be nice to work on it.